codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -430,27 +434,26 @@ let runLoaders = (
     // since we'll need to rerun each loader separately
     // before the handler, to get the uptodate entities from the in memory store.
     // Still need to propogate the errors.
-    let _: array<unknown> =
+    let _: array<Internal.loaderReturn> =
       await eventBatch
-      ->Array.keepMap(eventBatchQueueItem => {
-        eventBatchQueueItem.handlerRegister
-        ->Types.HandlerTypes.Register.getLoaderHandler
-        ->Option.map(
-          ({loader}) => {
-            let contextEnv = ContextEnv.make(~eventBatchQueueItem, ~logger)
+      ->Array.keepMap(eventItem => {
+        switch eventItem {
+          | {loader: Some(loader)} => {
+            let contextEnv = ContextEnv.make(~eventItem, ~logger)
```

Comment:
While I'm thinking about logging, I think the logger passed to this first run of the loaders should contain a note about the fact that it's an initial run of the loader and it will happen again. To help users understand if they are logging something in the loader why it might be twice.

Reply:
Sounds like a good idea.